packages/tools/request/src/fetchManager/README.md

```diff
@@ -0,0 +1,100 @@
+- `_shouldPrefetch`: boolean, if true downloads all resources specified in `_uriMap`, otherwise only those requested in `_asapKeys`.
+- `_uriMap`: Stream accepting a map of names to resources. Besides the required URI, each resource may specify a custom transformation function. `{[key]: {url: string, transformer?: () => {}}}` 
+
+## Output props
+
+- `_outData`: output stream for subscribing that receives the file contents: `{[key]: fileContents}`
+- `_outEvents`: output stream useful for tracking the state of the download process.
+- `_outLoadingKeys`: output stream with array of keys that are currently being loaded.
+
+## Effect of changing props during execution
+
+Changing `_uriMap` clears all cached files, aborts all running downloads and
+fetches all resources specified, even if they were previously downloaded.
+
+Changing `_asapKeys` or `_nextKeys` before all downloads complete will keep any
+running downloads that are still requested according to their priority and abort
+the rest. Any previously completed downloads will be maintained.
+
+Setting `_shouldPrefetch` to `false`, if previously `true`, will abort any running
+downloads except those specified in `_asapKeys`.
+
+## Usage
+
+```javascript
+import {createFetchManagerStreams}
+	from '@svizzle/request/src/fetchManager/fetchManager';
+import {makeWebStreamsFetcher} from '@svizzle/request/src/webstreams';
+
+// Use browser supplied `fetch` function. An alternative `fetch` implementation
+// for Node supporting Web Streams can be found in `undici` NPM package.
+const webStreamsFetcher = makeWebStreamsFetcher(fetch);
+
+const {
+	_asapKeys,
+	_nextKeys,
+	_outData,
+	_outEvents,
+	_outLoadingKeys,
+	_shouldPrefetch,
+	_uriMap,
+} = createFetchManagerStreams(webStreamsFetcher);
+
+_shouldPrefetch.next(true);
+_uriMap.next({
+	'FILE1':{
+		url: 'https://unpkg.com/@svizzle/atlas@0.6.0/data/dist/NUTS/topojson/NUTS_RG_10M_2013_4326_LEVL_0.json'
+	},
+	'FILE2':{
+		url: 'https://unpkg.com/@svizzle/atlas@0.6.0/data/dist/NUTS/topojson/NUTS_RG_10M_2013_4326_LEVL_1.json'
+	},
+	'FILE3':{
+		url: 'https://unpkg.com/@svizzle/atlas@0.6.0/data/dist/NUTS/topojson/NUTS_RG_10M_2013_4326_LEVL_2.json'
+	}
+});
+_asapKeys.next(['FILE3']);
+_nextKeys.next(['FILE2']);
+
+_outData.subscribe(data => {
+	if ('FILE1' in data) {
+		// FILE1 contents are completly loaded and stored in a byte array
+	}
+})
+```
+
+You may also specify a transformation function to postprocess file contents when
+a download completes.
+
+```javascript
+// ...
+const decoder = new TextDecoder('utf-8');
+const transformer = JSON.parse(decoder.decode(buffer));
+
+const webStreamsFetcher = makeWebStreamsFetcher(fetch, transformer);
+// ...
+```
+
+## Status
+
+As note for future developments, we agreed that this implementation is too
+brittle and that we should use a state machine to manage the logic rather than
+Rx streams.
```

Comment:
```suggestion
As a note for future developments, we agreed that this implementation is too
brittle and that we should use a state machine to manage the logic rather than
Rx streams.
```